uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/AbsencePeriod.php

```diff
@@ -618,4 +619,83 @@ public static function getPeriodsBetweenDates(DateTime $fromDate, DateTime $toDa
 
     return $absencePeriods;
   }
+
+  /**
+   * Returns the closest Absence Period to the current date.
+   * The absence period could be before the current date or
+   * could be after or could even contain the current date.
+   *
+   * @return AbsencePeriod|null
+   *   Returns null if there are no absence periods.
+   */
+  public static function getClosestToCurrentDate() {
+    $firstPeriod = self::getFirst();
```

Comment:
There are a few problems with this method:
- The description doesn't describe the exact behavior of the method. It doesn't mention that the method favors absence periods in the future, even if we have there is a period in the past closer to the current date.
- It uses this `getFirst()` method which is problematic, because it order things based on the weight but then all the comparisons as done by dates, and we can have a absence period which the dates are in the future but the weight is 1. In the future we'll probably be removing this weigh field, but for now we shouldn't use it for ordering like.
- The method is a bit complex and may result in a lot of queries being sent to the database. 

For the last query, I think we can probably do it with a single query. Something like this:

```php
$today = date('Y-m-d');
$query = "SELECT * FROM absence_period ORDER BY ABS( DATEDIFF(start_date, {$today} ) ), start_date DESC LIMIT 1";
```

I based that on: https://stackoverflow.com/questions/6186962/sql-query-to-show-nearest-date. The ABS() function is so that both -1 and 1 are considered the same and the ordering by start_date is to make sure that period starting in the future will have a higher precedence. Do you think this could work?

Reply:
I didn't know the weight field is not reliable as I based most of the logic on it. 

Went with two queries as agreed since the query above will not give preference to absence periods in the future.